src/main/java/org/opensearch/flowframework/workflow/CreateIndexStep.java

```diff
@@ -74,7 +74,7 @@ public void onFailure(Exception e) {
 
         for (WorkflowData workflowData : data) {
             Map<String, Object> content = workflowData.getContent();
-            index = (String) content.get("index-name");
+            index = (String) content.get("index_name");
```

Comment:
Do we have this field name in a constant somewhere? Let's use it (to avoid typos like the one you're fixing!)

Not part of your PR but the `settings != null` conditional below is never going to shortcut this loop as it's not ever fetched.  Is that missing in this iteration?

Reply:
Ill change these to constants and move them over to our Common class. As for the index settings, we still need to add support for that, but its not a priority at the moment